src/utils/validation.ts

```diff
@@ -2,4 +2,10 @@ function isValidEmail(email: string) {
   return /^\w+[\w-.]*@\w+((-\w+)|(\w*))\.[a-z]{2,3}$/.test(email)
 }
 
-export { isValidEmail }
+function isValidUrl(url: string) {
+  const urlRegex =
+    /^(https?:\/\/)((localhost)|((((25[0-5])|(2[0-4][0-9])|([0-1]?[0-9]?[0-9]))\.){3}((25[0-5])|(2[0-4][0-9])|([0-1]?[0-9]?[0-9])))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})))(?::\d{2,5})?(?:\/\S*)?$/
```

Comment:
This regex is ... _complicated_. 

I'm assuming this is coming from [this reference](https://www.freecodecamp.org/news/check-if-a-javascript-string-is-a-url/).

```
const isValidUrl = urlString=> {
	 var urlPattern = new RegExp('^(https?:\\/\\/)?'+ // validate protocol
	  '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ // validate domain name
	  '((\\d{1,3}\\.){3}\\d{1,3}))'+ // validate OR ip (v4) address
	  '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+ // validate port and path
	  '(\\?[;&a-z\\d%_.~+=-]*)?'+ // validate query string
	  '(\\#[-a-z\\d_]*)?$','i'); // validate fragment locator
	return !!urlPattern.test(urlString);
}
```

Please include these comments describing each piece if you prefer to continue using the regex. Otherwise, the article suggests a simpler implementation with `new URL`

```
function isValidUrl(url: string) {
  try {
    return Boolean(new URL(url))
  } catch () {
    return false;
  }
}
```

Reply:
Can this approach make it easy?  https://github.com/arcana-network/developer-dashboard/pull/128/commits/b6e9439b7e3f674b474026837488e9988260f9fd

Reply:
This looks a lot more readable now, thank you!